docs/topics/Voice_Connections.md

```diff
@@ -1,89 +1,189 @@
+	}
+}
+```
+
+>danger
+>`heartbeat_interval` here is an erroneous field and should be ignored. The correct heartbeat_interval value comes from the Hello payload.
+
+## Heartbeating
+
+In order to maintain your websocket connection, you need to continuously send heartbeats at the interval determined in [OP 8 Hello](#DOCS_VOICE_CONNECTIONS/voice-events-voice-op-codes):
+
+###### Example Hello Payload
+
+```json
+{
+	"op": 8,
+	"d": 41250
+}
+```
+
+This is sent immediately after [OP 2 Ready](#DOCS_VOICE_CONNECTIONS/voice-events-voice-op-codes). After this, you should send [OP 3 Heartbeat](#DOCS_VOICE_CONNECTIONS/voice-events-voice-op-codes)—which contains an integer nonce—every elapsed interval:
+
+###### Example Heartbeat Payload
+
+```json
+{
+	"op": 3,
+	"d": 1501184119561
+}
+```
+
+In return, you will be sent back an [OP 6 Heartbeat ACK](#DOCS_VOICE_CONNECTIONS/voice-events-voice-op-codes) that contains the previously sent nonce:
+
+###### Example Heartbeat ACK Payload
+
+```json
+{
+	"op": 6,
+	"d": 1501184119561
 }
 ```
 
-### Establishing a Voice UDP Connection
+## Establishing a Voice UDP Connection
 
-Once we receive the properties of a UDP voice server from our OP2 ready payload, we can proceed to the final step of voice connections, which entails establishing and handshaking a UDP connection for voice data. First, we open a UDP connection to the same endpoint we originally received in the [Voice Server Update](#DOCS_GATEWAY/voice-server-update) payload, combined with the port we received in the Voice Ready payload. If required, we can now perform an [IP Discovery](#DOCS_VOICE_CONNECTIONS/ip-discovery) using this connection. Once we've fully discovered our external IP and UDP port, we can then tell the voice websocket what it is, and start receiving/sending data. We do this using the select protocol payload.
+Once we receive the properties of a UDP voice server from our [OP 2 Ready](#DOCS_VOICE_CONNECTIONS/voice-events-voice-op-codes) payload, we can proceed to the final step of voice connections, which entails establishing and handshaking a UDP connection for voice data. First, we open a UDP connection to the same endpoint we originally received in the [Voice Server Update](#DOCS_GATEWAY/voice-server-update) payload, combined with the port we received in the Voice Ready payload. If required, we can now perform an [IP Discovery](#DOCS_VOICE_CONNECTIONS/ip-discovery) using this connection. Once we've fully discovered our external IP and UDP port, we can then tell the voice websocket what it is, and start receiving/sending data. We do this using [OP 1 Select Protocol](#DOCS_VOICE_CONNECTIONS/voice-events-voice-op-codes):
 
 >warn
-> The plain mode is deprecated and will be removed soon. All data should be sent using a supported encryption method (right now only `xsalsa20_poly1305`).
+>The plain mode is no longer supported. All data should be sent using a supported encryption method, right now only `xsalsa20_poly1305`.
```

Comment:
replied to b1nzy but moving it here.

lets just remove it imo.. it's been deprecated and is no longer supported.. so just undocument it heh